pkg/build/builder/source.go

```diff
@@ -56,38 +65,52 @@ func fetchSource(dir string, build *api.Build, urlTimeout time.Duration, in io.R
 	return nil
 }
 
+// checkRemoteGit validates the specified Git URL. It returns GitNotFoundError
+// when the remote repository not found and GitAuthenticationError when the
+// remote repository failed to authenticate.
+// Since this is calling the 'git' binary, the proxy settings should be
+// available for this command.
```

Comment:
I would like to see e2e tests for those cases

On Nov 12, 2015, at 5:56 AM, Ben Parees <notifications@github.com> wrote:

In pkg/build/builder/source.go
<https://github.com/openshift/origin/pull/5573#discussion_r44658788>:

> @@ -56,38 +65,52 @@ func fetchSource(dir string, build *api.Build, urlTimeout time.Duration, in io.R
>  	return nil
>  }
>
> +// checkRemoteGit validates the specified Git URL. It returns GitNotFoundError
> +// when the remote repository not found and GitAuthenticationError when the
> +// remote repository failed to authenticate.
> +// Since this is calling the 'git' binary, the proxy settings should be
> +// available for this command.

so i'm still a bit nervous about this, given how badly proxy behavior was
originally broken.

We need to make sure you or QE tests this behavior under the a scenario in
which the remote git server can only be accessed via the proxy (previously
both myself and QE only tested it under a condition in which the git server
could be reached whether or not you went through the proxy, and then setup
a broken proxy to confirm the requests were going through the proxy).

Alternatively you can test this by setting up a broken proxy and then
confirming that the checkRemoteGit call fails, since that would also
confirm the call is going through the proxy.

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/pull/5573/files#r44658788>.